src/PostgREST/Auth.hs

```diff
@@ -57,6 +64,15 @@ jwtClaims AppConfig{..} payload time = do
     jwtClaimsError JWT.JWTExpired = JwtTokenInvalid "JWT expired"
     jwtClaimsError e              = JwtTokenInvalid $ show e
 
+headerJWT :: Request -> Text
```

Comment:
I split this out of `ApiRequest` as it's a separate concern from all the other request parsing. This allows us to check the JWT before we even load the body of the request, which should improve DoS resistance a bit.

Reply:
Yes, that makes a lot of sense!

Reply:
Yes, that makes a lot of sense!